packages/lwc-language-server/src/javascript/compiler.ts

```diff
@@ -121,20 +124,43 @@ function toDiagnostic(err: any): Diagnostic {
 }
 
 export async function compileSource(source: string, fileName = 'foo.js'): Promise<CompilerResult> {
+    const name = fileName.substring(0, fileName.lastIndexOf('.'));
+
+    const transformOptions = {
+        name,
+        namespace: 'x',
+    };
     try {
-        const name = fileName.substring(0, fileName.lastIndexOf('.'));
-        const options: CompilerOptions = {
-            name,
-            namespace: 'x',
-            files: {},
-        };
-        const transformerResult = await transform(source, fileName, options);
-        const metadata = transformerResult.metadata as Metadata;
-        patchComments(metadata);
-        return { metadata, diagnostics: [] };
+        transformSync(source, fileName, transformOptions);
     } catch (err) {
```

Comment:
Some compilation errors will only appear upon an attempted compilation. These are generally restricted to non-syntax errors, such as the fact that boolean public properties cannot be true by default (otherwise, how would the parent component disable the flag?).

So, we transform first using the up-to-date compiler to ensure no compilation issues exist. If it succeeds, we collect metadata about the component code below.